### ISSUE TITLE
IL_INFINITE_LOOP cannot detect global variable sometimes

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1866]
Hi, I found that the rule [IL_INFINITE_LOOP](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#il-an-apparent-infinite-loop-il-infinite-loop) cannot detect global variables sometimes. Please refet to the following case:

```java
public class A {
    int result = 0;
    int sum(int n) {   
        for (int i = 0; i < n;)
            result += i;  // should report a warning for this loop
        return result;
    }
}
```
However, it is interesting that this rule can detect some cases related to global variable:
```java
public class A {
  int result = 0;
  int sumWhileDo(int n) {    
    int i = 0;
    while (i < n) {
        result += i;  // this loop can be detected
    }
    return result;
  }
}
```
**Expected Outcome:** A warning
**SpotBugs Version:** 4.5.0
